Software/Backend/DataAccessLayer/Infrastructure/WorkingRepository.cs

```diff
@@ -0,0 +1,101 @@
+﻿using DataAccessLayer.AppLogic;
+using DataAccessLayer.Models;
+using Microsoft.Data.SqlClient;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Infrastructure
+{
+    public class WorkingRepository : IWorkingRepository
+    {
+        private readonly DB _db;
+
+        public WorkingRepository(DB db)
+        {
+            _db = db;
+        }
+
+        public bool CallWorkerToFirstAvailableEntry(int workerId, int jobId, int skillId)
+        {
+            // Dohvati sve retke za zadani job_id i skill_id
+            var entries = GetWorkingEntriesByJobId(jobId);
+
+            if (entries == null || !entries.Any())
+            {
+                Console.WriteLine("Nema dostupnih redaka za zadani posao.");
+                return false;
+            }
+
+            // Uzmi prvi redak koji nema dodijeljen worker_id i ima odgovarajući skill_id
+            var entryToUpdate = entries.FirstOrDefault(e => e.WorkerId == null && e.SkillId == skillId);
+            if (entryToUpdate == null)
+            {
+                Console.WriteLine("Nema dostupnih redaka bez dodijeljenog radnika s odgovarajućim skill_id.");
+                return false;
+            }
```

Comment:
U komentaru si napisao da u GetWorkingEntriesByJobId dohvacas po skill_id sto nije istina kolko vidim nego samo po job_id pa ovdje kasnije onda uzimas prvi redak, mislim da bi mozda bilo bolje da to sve stavis u funkciju GetWorkingEntriesByJobId da ti ona automatski vraca prvi prazan.